From 6a19d94994fe7a9e4434cf3b6085b05fe54ba502 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Sun, 05 Jun 2022 15:59:34 +0300
Subject: [PATCH] Revert "[SW-82716] common: add crc8 support"

Revert rom recovery implementation for secured firmware update via
inband and OOB.

This reverts commit 1d217247da8f6248beb0221f63d4f162fafa0aeb.

Change-Id: I4b38c1e47a8577151bfcf1b18672cab9179adc11
---

diff --git a/common/crc8.c b/common/crc8.c
deleted file mode 100644
index 33b5ab2..0000000
--- a/common/crc8.c
+++ /dev/null
@@ -1,49 +0,0 @@
-// SPDX-License-Identifier:     GPL-2.0+
-/*
- * Copyright (C) 2022 HabanaLabs Ltd.
- * All Rights Reserved.
- *
- */
-
-#include <arcboot.h>
-#include <crc8.h>
-
-/* This code was copied from linux/lib/crc8.c */
-
-/**
- * crc8_populate_msb - fill crc table for given polynomial in reverse bit order.
- *
- * @table:	table to be filled.
- * @polynomial:	polynomial for which table is to be filled.
- */
-void crc8_populate_msb(uint8_t table[CRC8_TABLE_SIZE], uint8_t polynomial)
-{
-	int i, j;
-	const uint8_t msbit = 0x80;
-	uint8_t t = msbit;
-
-	table[0] = 0;
-
-	for (i = 1; i < CRC8_TABLE_SIZE; i *= 2) {
-		t = (t << 1) ^ (t & msbit ? polynomial : 0);
-		for (j = 0; j < i; j++)
-			table[i+j] = table[j] ^ t;
-	}
-}
-
-/**
- * crc8 - calculate a crc8 over the given input data.
- *
- * @table: crc table used for calculation.
- * @pdata: pointer to data buffer.
- * @nbytes: number of bytes in data buffer.
- * @crc: previous returned crc8 value.
- */
-uint8_t crc8(const uint8_t table[CRC8_TABLE_SIZE], uint8_t *pdata, size_t nbytes, uint8_t crc)
-{
-	/* loop over the buffer data */
-	while (nbytes-- > 0)
-		crc = table[(crc ^ *pdata++) & 0xff];
-
-	return crc;
-}
diff --git a/common/crc8.h b/common/crc8.h
deleted file mode 100644
index acd15ca..0000000
--- a/common/crc8.h
+++ /dev/null
@@ -1,16 +0,0 @@
-/* SPDX-License-Identifier:     GPL-2.0+
- *
- * Copyright (C) 2022 HabanaLabs Ltd.
- * All Rights Reserved.
- *
- */
-
-#ifndef _CRC8_H_
-#define _CRC8_H_
-
-#define CRC8_TABLE_SIZE		256
-
-void crc8_populate_msb(uint8_t table[CRC8_TABLE_SIZE], uint8_t polynomial);
-uint8_t crc8(const uint8_t table[CRC8_TABLE_SIZE], uint8_t *pdata, size_t nbytes, uint8_t crc);
-
-#endif
